### PR TITLE
feat: test packages separately

### PR DESCRIPTION
**Pull request by trevorwhitney**
This PR improves the `check` phase of our build (that runs for both releases and PRs) by ~4m, decreasing the current runtime of ~9m30s to ~5m30s. It does this by splitting out our tests in a job per package using GitHub actions 'matrix' job strategy.

This is what it will look like run on `grafana/loki`: https://github.com/grafana/loki/actions/runs/8726432895

### CODE REVIEW

File: workflows/validate.libsonnet [hunk 1]
```diff
@@ -35,44 +35,77 @@ local validationJob = _validationJob(false);
...
-    validationJob
-    + job.withSteps([
-      validationMakeStep('test', 'test'),
-    ])
-  ),
+  // Test jobs
+  collectPackages: job.new()
+                   + job.withSteps([
+                     common.checkout,
+                     common.fixDubiousOwnership,
```
Comment (cstyan) [inline workflows/validate.libsonnet:42]: lolwut

File: workflows/validate.libsonnet [hunk 2]
```diff
@@ -35,44 +35,77 @@ local validationJob = _validationJob(false);
...
+                     step.new('gather packages')
+                     + step.withId('gather-tests')
+                     + step.withRun(|||
+                       echo "packages=$(find . -path '*_test.go' -printf '%h\n' \
+                         | grep -e "pkg/push" -e "integration" -e "operator" -e "lambda-promtail" -v \
```
Comment (cstyan) [inline workflows/validate.libsonnet:47]: so to be clear, we're not running the integration or operator tests anymore? and pkg/push and lambda-promtail will be run as separate jobs?

**Review by cstyan — APPROVED** [review]

File: workflows/validate.libsonnet (hunk 2 above)
Comment (trevorwhitney) [reply workflows/validate.libsonnet:47]: not running operator, correct, it has it's own workflow that's guarded on changes to that folder. we are running integration tests just in a different, parallel job

**Comment by trevorwhitney** [comment]
going to merge monday in case there's fallout

**Pull request merged**